### PR TITLE
`eachAlive` Handlebars Helper 

### PR DESCRIPTION
**Pull request by smurawski**
to simplify the each/if pattern for supervisor templates.

To go from: 

Before

```
{{#if bind.has_backend }}
{{~#each bind.backend.members}}
{{~#if alive }}
server ip ip:port
{{~/if}}
{{~/each}}
{{~/if}}
```

```
{{#if bind.has_backend }}
{{~#eachAlive bind.backend.members}}
server ip ip:port
{{~/eachAlive}}
{{~/if}}
```

This will need a rebase on top of #1811 and needs docs yet.

Signed-off-by: Steven Murawski <steven.murawski@gmail.com>

### CODE REVIEW

**Comment by thesentinels** [comment]
Thanks for the pull request! Here is what will happen next:
 1. Your PR will be reviewed by the maintainers
 2. If everything looks good, one of them will `approve` it, and your PR will be merged.

Thank you for contributing!

File: components/sup/src/templating/helpers.rs [hunk 1]
```diff
@@ -14,16 +14,49 @@
...
 
+pub fn each_alive(h: &Helper, r: &Handlebars, rc: &mut RenderContext) -> RenderResult {
+    let value = try!(h.param(0)
+            .ok_or_else(|| RenderError::new("Param not found for helper \"eachAlive\"")))
+        .value();
+
+    let template = try!(h.template()
+        .ok_or_else(|| RenderError::new("No content to render inside the helper \"eachAlive\"")));
+    let default_array = Vec::default();
+    let value_array = value.as_array().unwrap_or(&default_array);
```
Comment (mwrock) [inline components/sup/src/templating/helpers.rs]: this is a total nit but an `if let` on `Ok` could avoid the need for `default_array`

**Comment by mwrock** [comment]
This looks great and +100 for submitting with a test! I guess we will let you rebase before unleashing the sentinals.

**Comment by mwrock** [comment]
Hmm or perhaps a rebase is unnecessary since GH shows no conflicts.

**Comment by mwrock** [comment]
@thesentinels approve

**Comment by thesentinels** [comment]
:metal: I am testing your branch against master before merging it. We do this to ensure that the master branch is never failing tests.

**Comment by thesentinels** [comment]
:neckbeard: Travis CI has [started testing this PR](https://travis-ci.org/habitat-sh/habitat/builds/204686237).

**Comment by thesentinels** [comment]
:neckbeard: Travis CI has [started testing this PR](https://travis-ci.org/habitat-sh/habitat/builds/204686237).

**Comment by thesentinels** [comment]
:sparkling_heart: Travis CI [reports this PR passed](https://travis-ci.org/habitat-sh/habitat/builds/204686237).

It always makes me feel nice when humans approve of one anothers work. I'm merging this PR now.

I just want you and the contributor to answer me one question:

gif-keyboard-3280869874741411265

**Pull request merged**